File: src/3146.permutation-difference-between-two-strings.3412/solution.ts
function findPermutationDifference (s: string, t: string): number {
    const positions = new Array(26);
    for (let i = 0; i < s.length; i++) {
        positions[s.charCodeAt(i) - 97] = i;
    }
    let result = 0;
    for (let i = 0; i < t.length; i++) {
        result += Math.abs(i - positions[t.charCodeAt(i) - 97]);
    }
    return result;
}

File: src/3146.permutation-difference-between-two-strings.3412/solution.java
class Solution {
    public int findPermutationDifference(String s, String t) {
        int[] positions = new int[26];
        char[] charsS = s.toCharArray();
        for(int i=0;i<charsS.length;i++){
            positions[charsS[i]-'a'] = i;
        }
        int result = 0;
        char[] charsT = t.toCharArray();
        for(int i=0;i<charsT.length;i++){
            result += Math.abs(i-positions[charsT[i]-'a']  );
        }
        return result;
    }
}